zrml/prediction-markets/src/migrations.rs

```diff
@@ -363,61 +284,187 @@ mod tests {
+            metadata: vec![4, 5],
+            market_type: MarketType::Categorical(14),
+            period: MarketPeriod::Block(6..7),
+            deadlines: Deadlines { grace_period: 8, oracle_duration: 9, dispute_duration: 10 },
+            scoring_rule: ScoringRule::CPMM,
+            status: MarketStatus::Disputed,
+            report: Some(Report { at: 11, by: 12, outcome: OutcomeReport::Categorical(13) }),
+            resolved_outcome: Some(OutcomeReport::Categorical(13)),
+            dispute_mechanism: mdm,
+        }
+    }
+
+    fn set_up_chain() {
+        StorageVersion::new(PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION).put::<Pallet<Runtime>>();
+    }
+}
+
+pub struct AddFieldToAuthorityReport<T>(PhantomData<T>);
+
+// Add resolve_at block number value field to `AuthorizedOutcomeReports` map.
+impl<T: Config + zrml_market_commons::Config + zrml_authorized::Config> OnRuntimeUpgrade
+    for AddFieldToAuthorityReport<T>
+{
+    fn on_runtime_upgrade() -> Weight
+    where
+        T: Config,
+    {
+        let mut total_weight = T::DbWeight::get().reads(1);
+        let authorized_version = StorageVersion::get::<AuthorizedPallet<T>>();
+        if authorized_version != AUTHORIZED_REQUIRED_STORAGE_VERSION {
+            log::info!(
+                "AddFieldToAuthorityReport: authorized version is {:?}, require {:?};",
+                authorized_version,
+                AUTHORIZED_REQUIRED_STORAGE_VERSION,
+            );
+            return total_weight;
+        }
+        log::info!("AddFieldToAuthorityReport: Starting...");
+
+        let mut new_storage_map = Vec::new();
+        let now = <frame_system::Pallet<T>>::block_number();
+        for (key, old_value) in
+            storage_iter::<Option<OutcomeReport>>(AUTHORIZED, AUTHORIZED_OUTCOME_REPORTS)
+        {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+            if let Some(outcome) = old_value {
+                let resolve_at: T::BlockNumber =
+                    now.saturating_add(<T as zrml_authorized::Config>::ReportPeriod::get());
+                let new_value = AuthorityReport { resolve_at, outcome };
+                new_storage_map.push((key, new_value));
+            }
+        }
+
+        for (key, new_value) in new_storage_map {
+            put_storage_value::<Option<AuthorityReport<T::BlockNumber>>>(
+                AUTHORIZED,
+                AUTHORIZED_OUTCOME_REPORTS,
+                &key,
+                Some(new_value),
             );
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
+
+        StorageVersion::new(AUTHORIZED_NEXT_STORAGE_VERSION).put::<AuthorizedPallet<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        log::info!("AddFieldToAuthorityReport: Done!");
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        Ok(())
```

Comment:
I would suggest to use set_temp_storage and get_temp_storage api as used in https://github.com/zeitgeistpm/zeitgeist/pull/778/files#diff-ad1ca6a081baba38d5273b6a485f1b546093cbec65939f63a2cf00797bc3814dR131 to have pre_upgrade() and post_upgrade() verify values.

Reply:
7c7098cf555572d1ab2e63d04cb3d18720ce7a64